airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/integration/test_report_based_streams.py

```diff
@@ -0,0 +1,266 @@
+            with self.subTest(msg=stream_name):
+                _mock_auth(http_mocker)
+
+                http_mocker.post(_create_report_request(stream_name).build(), _create_report_response())
+                http_mocker.get(_check_report_status_request(_REPORT_ID).build(), _check_report_status_response(stream_name))
+                http_mocker.get(
+                    _get_document_download_url_request(_REPORT_DOCUMENT_ID).build(), _get_document_download_url_response()
+                )
+                http_mocker.get(
+                    _download_document_request(_DOCUMENT_DOWNLOAD_URL).build(),
+                    _download_document_response(stream_name, data_format=params.get("data_format", "csv")),
+                )
+
+                output = self._read(stream_name, config())
+                assert len(output.records) == 2
+
+    @HttpMocker()
+    def test_given_compressed_report_when_read_then_return_records(self, http_mocker: HttpMocker) -> None:
+        for stream_name, params in STREAMS.items():
+            with self.subTest(msg=stream_name):
+                _mock_auth(http_mocker)
+
+                http_mocker.post(_create_report_request(stream_name).build(), _create_report_response())
+                http_mocker.get(_check_report_status_request(_REPORT_ID).build(), _check_report_status_response(stream_name))
+                http_mocker.get(
+                    _get_document_download_url_request(_REPORT_DOCUMENT_ID).build(),_get_document_download_url_response(compressed=True)
+                )
+
+                # a workaround to pass compressed document to the mocked response
+                document_request = _download_document_request(_DOCUMENT_DOWNLOAD_URL).build()
+                document_response = _download_document_response(stream_name, data_format=params.get("data_format", "csv"), compressed=True)
+                document_request_matcher = HttpRequestMatcher(document_request, minimum_number_of_expected_match=1)
+                http_mocker._matchers.append(document_request_matcher)
+
+                http_mocker._mocker.get(
+                    requests_mock.ANY,
+                    additional_matcher=http_mocker._matches_wrapper(document_request_matcher),
+                    response_list=[{"content": document_response.body, "status_code": document_response.status_code}],
+                )
+
+                output = self._read(stream_name, config())
+                assert len(output.records) == 2
+
+    @HttpMocker()
+    def test_given_report_access_forbidden_when_read_then_no_records(self, http_mocker: HttpMocker) -> None:
+        for stream_name in STREAMS.keys():
+            with self.subTest(msg=stream_name):
+                _mock_auth(http_mocker)
+
+                http_mocker.post(_create_report_request(stream_name).build(), _create_report_response(status_code=403))
+
+                output = self._read(stream_name, config())
+                assert len(output.records) == 0
```

Comment:
It feels like this shouldn't be the outcome of the test but rather that a config_error is returned

Reply:
In the current implementation, there's a warning log message. Added this assertion to the test

Reply:
Initially this implementation was a sort of replacement for the availability strategy as the default approach we use in the `HttpAvailabilityStrategy` wouldn't work here because some reports become unavailable for some period after it was triggered to create. From what I found [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py#L121), when the stream is not available, the source just logs the message and skips reading for this stream, but I didn't find any status change there. Could you please advice where can I find this logic to review it and update the test accordingly?

Reply:
I don't know enough about Amazon Seller Partner and the availability and what is the expectation for the user. If a stream is unavailable, how is it shown to the user? Can someone take action? Should they receive an email to be notified to do something?

Reply:
As I know, in current implementation we just log the warning message even if we use some availability strategy, and there's no other way for the user to know about that except reading logs

Reply:
Ok! If you think this is important, we should create a ticket. I'm fine with the test reflecting that for now. Thanks Anton!